query-service-impl/src/main/java/org/hypertrace/core/query/service/postgres/PostgresClientFactory.java

```diff
@@ -58,78 +62,60 @@ public PostgresClient getPostgresClient(String clusterName) {
+      poolableConnectionFactory.setDefaultAutoCommit(false);
+      poolableConnectionFactory.setDefaultTransactionIsolation(
+          Connection.TRANSACTION_READ_COMMITTED);
+      poolableConnectionFactory.setPoolStatements(false);
+      return new PoolingDataSource<>(connectionPool);
     }
 
-    public ResultSet executeQuery(String statement, Params params) throws SQLException {
-      PreparedStatement preparedStatement = buildPreparedStatement(statement, params);
-      return preparedStatement.executeQuery();
+    private PostgresClient(DataSource dataSource) {
+      this.dataSource = dataSource;
     }
 
-    private PreparedStatement buildPreparedStatement(String statement, Params params)
-        throws SQLException {
-      String resolvedStatement = resolveStatement(statement, params);
-      return connection.prepareStatement(resolvedStatement);
+    public Connection getConnection() throws SQLException {
+      return dataSource.getConnection();
     }
 
-    @VisibleForTesting
-    /*
-     * Postgres PreparedStatement creates invalid query if one of the parameters has '?' in its value.
-     * Sample postgres query : select * from table where team in (?, ?, ?).. Now say parameters are:
-     * 'abc', 'pqr with (?)' and 'xyz'..
-     *
-     * Now, on executing PreparedStatement:fillStatementWithParameters method on this will return
-     * select * from table where team in ('abc', 'pqr with ('xyz')', ?) -- which is clearly wrong
-     * (what we wanted was select * from table where team in ('abc', 'pqr with (?)', 'xyz'))..
-     *
-     * The reason is the usage of replaceFirst iteration in the postgres PreparedStatement method..
-     *
-     * Hence written this custom method to resolve the query statement rather than relying on Postgres's
-     * library method.
-     * This is a temporary fix and can be reverted when the Postgres issue gets resolved
-     * Raised an issue in incubator-postgres github repo: apache/incubator-postgres#6834
-     */
-    static String resolveStatement(String query, Params params) {
-      if (query.isEmpty()) {
-        return query;
-      }
-      String[] queryParts = query.split("\\?");
-
-      String[] parameters = new String[queryParts.length];
-      params.getStringParams().forEach((i, p) -> parameters[i] = getStringParam(p));
-      params.getIntegerParams().forEach((i, p) -> parameters[i] = String.valueOf(p));
-      params.getLongParams().forEach((i, p) -> parameters[i] = String.valueOf(p));
-      params.getDoubleParams().forEach((i, p) -> parameters[i] = String.valueOf(p));
-      params.getFloatParams().forEach((i, p) -> parameters[i] = String.valueOf(p));
-
-      StringBuilder sb = new StringBuilder();
-      for (int i = 0; i < queryParts.length; i++) {
-        sb.append(queryParts[i]);
-        sb.append(parameters[i] != null ? parameters[i] : "");
+    private int getPercentOf(int maxConnections, int percent) {
+      int value = (maxConnections * 100) / percent;
```

Comment:
For `maxConnection` as `5` this will be `25` based on above method call - `getPercentOf(maxConnections, 20)`.

Shall we do with ?
```
Math.floor(maxConnections * percent/100) 
```

Reply:
Fixed the calculation. Don't need to use floor function.